src/DictVectors/communicators.jl

```diff
@@ -308,6 +350,231 @@ function copy_to_local!(ptp::PointToPoint, w, t)
         end
     end
 
-    # Pack the segments into a PDVec and return it.
+    return main_column(w)
+end
+
+"""
+    NestedSegmentedBuffer{T}(nrows) <: AbstractMatrix{AbstractVector{T}}
+
+Matrix of vectors stored in a single buffer with collective MPI communication support. The
+number of rows in the matrix `nrows`. Used in the [`AllToAll`](@ref) communication strategy.
```

Comment:
```suggestion
Matrix of vectors stored in a single buffer with collective MPI communication support. `nrows` is the
number of rows in the matrix. Used in the [`AllToAll`](@ref) communication strategy.
```

Should it be explained what the rows and columns mean in the context of `AllToAll`?